OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameMenuLogic.cs

```diff
@@ -88,14 +88,48 @@ public IngameMenuLogic(Widget widget, World world, Action onExit, WorldRenderer
 
 			abortMissionButton.OnClick = () =>
 			{
-				if (world.IsGameOver)
+				hideMenu = true;
+				Action restartAction = null;
+				if (world.IsReplay || world.Map.Visibility == MapVisibility.MissionSelector)
 				{
-					onQuit();
-					return;
+					var iop = world.WorldActor.TraitsImplementing<IObjectivesPanel>().FirstOrDefault();
+					var exitDelay = iop != null ? iop.ExitDelay : 0;
+					restartAction = () =>
+					{
+						Ui.CloseWindow();
+						if (mpe != null)
+						{
+							if (Game.IsCurrentWorld(world))
+								mpe.Fade(MenuPaletteEffect.EffectType.Black);
+							exitDelay += 40 * mpe.Info.FadeLength;
```

Comment:
It is definitely ticks and should use gamespeed, but that can be left for another PR because your reasoning for not changing it is valid.

[Reference](https://github.com/OpenRA/OpenRA/blob/da4f04acfb2983038d07baa52d1cc42fcfafcd31/OpenRA.Mods.Common/Traits/PaletteEffects/MenuPaletteEffect.cs#L21-L22).